main.tf

```diff
@@ -139,7 +139,7 @@ resource "aws_route" "public" {
 resource "aws_route_table_association" "public" {
   count          = local.public_subnets
   subnet_id      = aws_subnet.public[count.index].id
-  route_table_id = aws_route_table.public[0].id
+  route_table_id = var.additional_route_tables ? element(aws_route_table.public[*].id, count.index) : aws_route_table.public[0].id
```

Comment:
This will need to be inverted as now the default is `true` instead of `false`.

Reply:
Why is the default set to true ? when not declared or if another module is reference we want this set to false

Reply:
suggestion is based on above logic of renaming the variable to 'shared_public_route_table'. Which would default to true, then the items after the ? would need to reverse order. 

reason is clarity. 'additional_route_tables' could apply to 'any route tables' but actually we can only apply this to the public route table(s).

Reply:
updated and tested against , true, false and not set